packages/itwinui-react/src/core/Dialog/DialogMain.tsx

```diff
@@ -204,34 +234,16 @@ export const DialogMain = React.forwardRef((props, ref) => {
   );
 
   return (
-    <Transition
-      in={isOpen}
-      appear={true}
-      timeout={{ exit: 600 }}
-      // Focuses dialog when opened
-      onEntered={() => {
-        previousFocusedElement.current = dialogRef.current?.ownerDocument
-          .activeElement as HTMLElement;
-        setFocus && dialogRef.current?.focus({ preventScroll: true });
-      }}
-      // Brings back focus to the previously focused element when closed
-      onExit={() => {
-        if (
-          dialogRef.current?.contains(
-            dialogRef.current?.ownerDocument.activeElement,
-          )
-        ) {
-          previousFocusedElement.current?.focus();
-        }
+    <DialogMainContext.Provider
+      value={{
+        beforeClose,
       }}
```

Comment:
nit 1: Context values should be memoized.

```suggestion
      value={React.useMemo(() => ({ beforeClose }), [beforeClose])}
```

(check formatting when committing)